robottelo/entities.py

```diff
@@ -418,6 +419,143 @@ def create_payload(self):
+    name = entity_fields.StringField(required=True, str_type='alphanumeric')
+    organization = entity_fields.OneToManyField('Organization', null=True)
+    tty = entity_fields.BooleanField(null=True)
+
+    class Meta(object):
+        """Non-field information about this entity."""
+        api_path = 'docker/api/v2/containers'
+        server_modes = ('sat')
+
+    def path(self, which=None):
+        """Extend :meth:`robottelo.orm.Entity.path`.
+
+        The format of the returned path depends on the value of ``which``:
+
+        logs
+            /content_view_versions/<id>/logs
+        power
+            /content_view_versions/<id>/power
+
+        ``super`` is called otherwise.
+
+        """
+        if which in ('logs', 'power'):
+            return '{0}/{1}'.format(
+                super(AbstractDockerContainer, self).path(which='self'),
+                which
+            )
+        return super(AbstractDockerContainer, self).path(which)
+
+    # NOTE: See BZ 1151220
+    def create_payload(self):
+        """Wrap submitted data within an extra dict."""
+        return {
+            u'container': super(AbstractDockerContainer, self).create_payload()
+        }
+
+    def read(self, auth=None, entity=None, attrs=None, ignore=()):
+        """Compensate for the unusual format of responses from the server.
+
+        The server returns an ID and a list of IDs for the ``compute_resource``
+        field. Compensate for this unusual design trait. Typically, the server
+        returns a hash or a list of hashes for ``OneToOneField`` and
+        ``OneToManyField`` fields.
+
+        """
+        if attrs is None:
+            attrs = self.read_json(auth)
+        compute_resource_id = attrs.pop('compute_resource_id')
+        if compute_resource_id is None:
+            attrs['compute_resource'] = None
+        else:
+            attrs['compute_resource'] = {'id': compute_resource_id}
+        return super(AbstractDockerContainer, self).read(
+            auth, entity, attrs, ignore
+        )
+
+    def power(self, power_action):
+        """Run a power operation on a container.
+
+        :param str power_action: One of 'start', 'stop' or 'status'.
+        :returns: Information about the current state of the container.
+        :rtype: dict
+
+        """
+        power_actions = ('start', 'stop', 'status')
+        if power_action not in power_actions:
+            raise ValueError('Received {0} but expected one of {1}'.format(
+                power_action, power_actions
+            ))
+        response = client.put(
+            self.path(which='power'),
+            auth=get_server_credentials(),
+            verify=False,
+            data={u'power_action': power_action},
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def logs(self, stdout=None, stderr=None, tail=None):
+        """Get logs from this container.
+
+        :param bool stdout: ???
```

Comment:
Not a blocker but let's have someone from the Docker Feature team look at this and give us some info?